R/ivreg.marg.R

```diff
@@ -0,0 +1,160 @@
+#' @export
+#' @rdname marg
+#' @examples
+#'
+#' # --------------------------------------
+#' # 2SLS using AER::ivreg (example adapted from ?AER::ivreg)
+#'
+#' data(margex)
+#' margex$assign <- margex$treatment
+#'
+#' # Probability of getting treatment increases with age and distance
+#' margex$pr_treat <- plogis(
+#'   margex$age / sd(margex$age) +
+#'   margex$distance / sd(margex$distance))
+#'
+#' # One-way non-compliance
+#' margex$actual <- margex$assign
+#' margex$actual[margex$assign == 1] <- rbinom(
+#'   n = sum(margex$assign == 1), size = 1,
+#'   prob = margex$pr_treat[margex$assign == 1]
+#' )
+#'
+#' mod <- AER::ivreg(y ~ as.factor(actual) + age + distance |
+#'                     as.factor(assign) + age + distance,
+#'                   data = margex)
+#'
+#'
+#' v <- function(object, ...){
+#'   vcov(object) * object$df.residual / nobs(object)
+#' }
+#'
+#' summary(mod, vcov = v)
+#'
+#' # Get the level of the outcome variable at different values of `actual`
+#' # Note data is a *required* parameter here
+#' marg(mod, var_interest = 'actual', data = margex, type = 'levels')
```

Comment:
In the `summary` function above, you use a custom `vcov` but not when you run `marg`. Why?